File: tixtix-app/src/main/java/com/example/TixTixController.java
package com.example;

import com.example.dto.*;
import org.springframework.web.bind.annotation.*;

import java.util.List;

@RestController
class TixTixController {
    private final TixTixService service;

    TixTixController(final TixTixService service) {
        this.service = service;
    }

    @GetMapping("/ping")
    String ping() {
        return "pong";
    }

    @PostMapping("/ticketing")
    TicketingResponse ticketing(@RequestBody final TicketingRequest request) {
        return service.ticketing(request);
    }

    @GetMapping("/{performanceId}")
    List<Long> getPerformance(@PathVariable final long performanceId) {
        return service.monitoring(performanceId);
    }

    @PostMapping("/bulk-ticketing")
    BulkTicketingResponse bulkTicketing(@RequestBody final List<BulkTicketingRequest> requests) {
        return service.bulkTicketing(requests);
    }

    @PostMapping("/sold-out-ticketing")
    void soldOutTicketing(@RequestBody final List<SoldOutTicketingRequest> requests) {
        service.soldOutTicketing(requests);
    }

}

File: tixtix-app-kot/src/main/kotlin/com/example/TixTixController.kt
package com.example

import com.example.dto.TicketingRequestDto
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.RequestBody
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController

@RestController
@RequestMapping("/tix")
class TixTixController(
    val service: TixTixService
) {

    @GetMapping("/ping")
    fun ping(): String = "pong 퐁퐁"

    @PostMapping()
    fun ticketing(@RequestBody requestDto: TicketingRequestDto): String = service.ticketing(requestDto)

}